### PR TITLE
Issue#544

### PR DESCRIPTION
**Pull request by pablohpsilva**
PR to fix issue #544.

It was added a `debounce` in order to hold back the event emission by either md-input and md-textarea.

Debounce is a prop and it's default value is: 300ms

After debouncing the `change` and `input` event emission, the lag went away.

Example:

issue544

### CODE REVIEW

**Comment by Samuell1** [comment]
Nice! I think maybe lower that 500ms to 100ms or 200ms

**Comment by pablohpsilva** [comment]
As default? Hum... OK! debounce is a prop, so you can config is as you want LOL. 

Is there a way to get in touch with @marcosmoura ?

**Comment by Samuell1** [comment]
@pablohpsilva btw its tested with select and options too ?

File: src/components/mdInputContainer/common.js [hunk 1]
```diff
@@ -29,6 +39,15 @@ export default {
       this.parentContainer.enableCounter = this.maxlength > 0;
       this.parentContainer.counterLength = this.maxlength;
     },
+    lazyEventEmitter() {
```
Comment (Morgul) [inline src/components/mdInputContainer/common.js]: I'm not sure why this is in `watch`? Shouldn't it be in `methods`?

Comment (pablohpsilva) [inline src/components/mdInputContainer/common.js]: I'm not sure why is github showing like that, but if you check the raw code, you'll see it is in methods. I don't know what sorcery happened here.

Comment (Morgul) [inline src/components/mdInputContainer/common.js]: Ah, ok. Yeah, that is weird.

**Comment by Samuell1** [comment]
@pablohpsilva any updates on this to merge? it will help performance @marcosmoura

**Comment by pablohpsilva** [comment]
Not yet @Samuell1 . @marcosmoura said he'll be back in a few days :D

**Comment by DavidBradbury** [comment]
Any updates?

**Comment by Morgul** [comment]
@DavidBradbury https://github.com/marcosmoura/vue-material/issues/598#issuecomment-285890369

**Comment by Samuell1** [comment]
@DavidBradbury if you want you can use my fork, i have some PRs in my fork to fix some things

**Comment by marcosmoura** [comment]
I'll approve that and then I will work on further improvements

**Pull request merged**